staging/src/k8s.io/client-go/tools/cache/listwatch.go

```diff
@@ -81,9 +81,21 @@ type Getter interface {
 	RESTClients() []restclient.Interface
 }
 
+// For backwards compatibility concerns, if the lister/watcher operates across all namespaces (namely cluster-scoped),
+// we should let it work on all tenants.
+// If it targets a specific namespaces, we should let it target at the system tenant.
+func InferTenantFromNamespace(namespace string) string {
```

Comment:
Why namespaceALL will lead to tenantALL?  A regular tenant user can still list/watch for namespaceALL, but it should be limited to his own tenant space, right?

Reply:
And we can't assume the default tenant is system tenant.

Reply:
Per offline discussion, we understand that this change unblocks the perf tests but it may have some issue for regular tenants. We cannot assume the tenant is "system" when the tenant info is not given.

Discussion results are recorded in https://github.com/futurewei-cloud/arktos/issues/515.